### PR TITLE
Added Log events for Httpsys

### PR DESCRIPTION
**Pull request by kuns200**
Added Log events for Httpsys
Also, Added event which were evented as 0.

Addresses #6381(in this specific format)

### CODE REVIEW

File: src/Servers/HttpSys/src/LoggerEventIds.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        public static EventId ErrorInListen = new EventId(5, "ErrorInListen");
+        public static EventId RequestProcessFailed = new EventId(6, "RequestProcessFailed");
+        public static EventId RequestEmptied = new EventId(7, "RequestEmptied");
+        public static EventId RequestCancelled = new EventId(8, "RequestCancelled");
+        public static EventId RequestStopped = new EventId(9, "RequestStopped");
+        public static EventId TokenError = new EventId(10, "TokenError");
+        public static EventId TokenDisconnect = new EventId(11, "TokenDisconnect");
+        public static EventId WaitTokenClosed = new EventId(12, "WaitTokenClosed");
+        public static EventId TokenDisconnectError= new EventId(13, "TokenDisconnectError");
+        public static EventId TokenDisconnectTriggered = new EventId(13, "TokenDisconnectTriggered");
```
Comment (Kahbazi) [inline src/Servers/HttpSys/src/LoggerEventIds.cs]: There are duplicate events for event id 13, 14 and 1.

File: src/Servers/HttpSys/src/MessagePump.cs [hunk 2]
```diff
@@ -274,7 +274,7 @@ void RegisterCancelation()
                 {
                     if (Interlocked.Exchange(ref _shutdownSignalCompleted, 1) == 0)
                     {
-                        _logger.LogInformation("Canceled, terminating " + _outstandingRequests + " request(s).");
+                        _logger.LogInformation(LoggerEventIds.RequestCancelled, "Canceled, terminating " + _outstandingRequests + " request(s).");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/MessagePump.cs]:
```suggestion
                        _logger.LogInformation(LoggerEventIds.StopCancelled, "Canceled, terminating " + _outstandingRequests + " request(s).");
```

File: src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs [hunk 3]
```diff
@@ -71,7 +71,7 @@ internal void SetProperty(HttpApiTypes.HTTP_SERVER_PROPERTY property, IntPtr inf
 
         internal void RegisterPrefix(string uriPrefix, int contextId)
         {
-            _logger.LogDebug("Listening on prefix: " + uriPrefix);
+            _logger.LogDebug(LoggerEventIds.PrefixListened, "Listening on prefix: {0} " , uriPrefix);
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs]:
```suggestion
            _logger.LogDebug(LoggerEventIds.RegisteringPrefix, "Listening on prefix: {0} " , uriPrefix);
```

File: src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs [hunk 4]
```diff
@@ -90,7 +90,7 @@ internal void RegisterPrefix(string uriPrefix, int contextId)
 
         internal bool UnregisterPrefix(string uriPrefix)
         {
-            _logger.LogInformation("Stop listening on prefix: " + uriPrefix);
+            _logger.LogInformation(LoggerEventIds.PrefixListenStopped, "Stop listening on prefix: {0} " , uriPrefix);
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs]:
```suggestion
            _logger.LogInformation(LoggerEventIds.UnregisteringPrefix, "Stop listening on prefix: {0} " , uriPrefix);
```

File: src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs [hunk 5]
```diff
@@ -117,7 +117,7 @@ public void Dispose()
 
             if (statusCode != UnsafeNclNativeMethods.ErrorCodes.ERROR_SUCCESS)
             {
-                _logger.LogError("CleanupV2Config; Result: " + statusCode);
+                _logger.LogError(LoggerEventIds.V2ConfigCleanupFailed, "CleanupV2Config; Result: {0}" , statusCode);
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs]:
```suggestion
                _logger.LogError(LoggerEventIds.CloseUrlGroupError, "HttpCloseUrlGroup; Result: {0}" , statusCode);
```

File: src/Servers/HttpSys/src/RequestProcessing/Request.cs [hunk 6]
```diff
@@ -338,11 +338,11 @@ public X509Certificate2 ClientCertificate
                     }
                     catch (CryptographicException ce)
                     {
-                        RequestContext.Logger.LogDebug(ce, "An error occurred reading the client certificate.");
+                        RequestContext.Logger.LogDebug(LoggerEventIds.CryptoException, ce, "An error occurred reading the client certificate.");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/Request.cs]:
```suggestion
                        RequestContext.Logger.LogDebug(LoggerEventIds.ErrorInReadingCertificate, ce, "An error occurred reading the client certificate.");
```

File: src/Servers/HttpSys/src/RequestProcessing/RequestContext.cs [hunk 7]
```diff
@@ -120,14 +120,14 @@ internal bool TryGetChannelBinding(ref ChannelBinding value)
...
-                Logger.LogDebug("TryGetChannelBinding; Channel binding requires HTTPS.");
+                Logger.LogDebug(LoggerEventIds.ChannelBindingNeedsHttps,"TryGetChannelBinding; Channel binding requires HTTPS.");
                 return false;
             }
 
             value = ClientCertLoader.GetChannelBindingFromTls(Server.RequestQueue, Request.UConnectionId, Logger);
 
             Debug.Assert(value != null, "GetChannelBindingFromTls returned null even though OS supposedly supports Extended Protection");
-            Logger.LogInformation("Channel binding retrieved.");
+            Logger.LogInformation(LoggerEventIds.ChannelBindingRetrived,"Channel binding retrieved.");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/RequestContext.cs]:
```suggestion
            Logger.LogDebug(LoggerEventIds.ChannelBindingRetrived,"Channel binding retrieved.");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs [hunk 8]
```diff
@@ -175,14 +175,14 @@ private unsafe void FlushInternal(bool endOfRequest, ArraySegment<byte> data = n
                 if (ThrowWriteExceptions)
                 {
                     var exception = new IOException(string.Empty, new HttpSysException((int)statusCode));
-                    Logger.LogError(0, exception, "Flush");
+                    Logger.LogError(LoggerEventIds.WriteFlushed, exception, "Flush");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs]:
```suggestion
                    Logger.LogError(LoggerEventIds.WriteError, exception, "Flush");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs [hunk 9]
```diff
@@ -175,14 +175,14 @@ private unsafe void FlushInternal(bool endOfRequest, ArraySegment<byte> data = n
...
                     Abort();
                     throw exception;
                 }
                 else
                 {
                     // Abort the request but do not close the stream, let future writes complete silently
-                    Logger.LogDebug($"Flush; Ignored write exception: {statusCode}");
+                    Logger.LogDebug(LoggerEventIds.WriteFlushedIgnored, $"Flush; Ignored write exception: {statusCode}");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs]:
```suggestion
                    Logger.LogDebug(LoggerEventIds.WriteErrorIgnored, $"Flush; Ignored write exception: {statusCode}");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs [hunk 10]
```diff
@@ -355,21 +355,21 @@ private unsafe Task FlushInternalAsync(ArraySegment<byte> data, CancellationToke
...
-                    Logger.LogError(0, exception, "FlushAsync");
+                    Logger.LogError(LoggerEventIds.ErrorWhenFlushAsync, exception, "FlushAsync");
                     Abort();
                     throw exception;
                 }
                 else
                 {
                     // Abort the request but do not close the stream, let future writes complete silently
-                    Logger.LogDebug($"FlushAsync; Ignored write exception: {statusCode}");
+                    Logger.LogDebug(LoggerEventIds.AbortButDonotClose,$"FlushAsync; Ignored write exception: {statusCode}");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs]:
```suggestion
                    Logger.LogDebug(LoggerEventIds.WriteErrorIgnored,$"FlushAsync; Ignored write exception: {statusCode}");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs [hunk 11]
```diff
@@ -649,21 +649,21 @@ internal unsafe Task SendFileAsyncCore(string fileName, long offset, long? count
...
-                    Logger.LogError(0, exception, "SendFileAsync");
+                    Logger.LogError(LoggerEventIds.FileSendAsyncFailed, exception, "SendFileAsync");
                     Abort();
                     throw exception;
                 }
                 else
                 {
                     // Abort the request but do not close the stream, let future writes complete silently
-                    Logger.LogDebug($"SendFileAsync; Ignored write exception: {statusCode}");
+                    Logger.LogDebug(LoggerEventIds.FileSendAsyncAbortButDonotClose,$"SendFileAsync; Ignored write exception: {statusCode}");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs]:
```suggestion
                    Logger.LogDebug(LoggerEventIds.FileSendAsyncErrorIgnored,$"SendFileAsync; Ignored write exception: {statusCode}");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs [hunk 12]
```diff
@@ -233,18 +233,18 @@ private static void IOCompleted(ResponseStreamAsyncResult asyncResult, uint erro
                 {
                     if (asyncResult._cancellationToken.IsCancellationRequested)
                     {
-                        logger.LogDebug($"FlushAsync.IOCompleted; Write cancelled with error code: {errorCode}");
+                        logger.LogDebug(LoggerEventIds.IOCompletedCancelled,$"FlushAsync.IOCompleted; Write cancelled with error code: {errorCode}");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs]:
```suggestion
                        logger.LogDebug(LoggerEventIds.WriteCancelled,$"FlushAsync.IOCompleted; Write cancelled with error code: {errorCode}");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs [hunk 13]
```diff
@@ -233,18 +233,18 @@ private static void IOCompleted(ResponseStreamAsyncResult asyncResult, uint erro
...
                         asyncResult.Cancel(asyncResult._responseStream.ThrowWriteExceptions);
                     }
                     else if (asyncResult._responseStream.ThrowWriteExceptions)
                     {
                         var exception = new IOException(string.Empty, new HttpSysException((int)errorCode));
-                        logger.LogError(0, exception, "FlushAsync.IOCompleted");
+                        logger.LogError(LoggerEventIds.IOCompletedCancelled, exception, "FlushAsync.IOCompleted");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs]:
```suggestion
                        logger.LogError(LoggerEventIds.WriteError, exception, "FlushAsync.IOCompleted");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs [hunk 14]
```diff
@@ -233,18 +233,18 @@ private static void IOCompleted(ResponseStreamAsyncResult asyncResult, uint erro
...
                         asyncResult.Fail(exception);
                     }
                     else
                     {
-                        logger.LogDebug($"FlushAsync.IOCompleted; Ignored write exception: {errorCode}");
+                        logger.LogDebug(LoggerEventIds.IOCompletedFailQuiet, $"FlushAsync.IOCompleted; Ignored write exception: {errorCode}");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs]:
```suggestion
                        logger.LogDebug(LoggerEventIds.WriteErrorIgnored, $"FlushAsync.IOCompleted; Ignored write exception: {errorCode}");
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs [hunk 15]
```diff
@@ -267,7 +267,7 @@ private static void IOCompleted(ResponseStreamAsyncResult asyncResult, uint erro
             }
             catch (Exception e)
             {
-                logger.LogError(0, e, "FlushAsync.IOCompleted");
+                logger.LogError(LoggerEventIds.IOCompletedFailed, e, "FlushAsync.IOCompleted");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs]:
```suggestion
                logger.LogError(LoggerEventIds.WriteError, e, "FlushAsync.IOCompleted");
```

**Review by Tratcher — CHANGES_REQUESTED** [review]

**Comment by kuns200** [comment]
i will fix them. thanks for your comments.

**Comment by kuns200** [comment]
fix in progress.

File: src/Servers/HttpSys/src/HttpSysListener.cs [hunk 16]
```diff
@@ -177,7 +177,7 @@ public void Start()
                     // Make sure the HttpListener instance can't be used if Start() failed.
                     _state = State.Disposed;
                     DisposeInternal();
-                    Logger.LogError(0, exception, "Start");
+                    Logger.LogError(LoggerEventIds.ListenerStartException, exception, "Start");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysListener.cs]: Be consistent about Error vs Exception vs Failed/Failure. Let's use Error as it's simpler.

File: src/Servers/HttpSys/src/HttpSysListener.cs [hunk 17]
```diff
@@ -238,14 +238,14 @@ private void Dispose(bool disposing)
...
-                    Logger.LogTrace("Disposing the listener.");
+                    Logger.LogTrace(LoggerEventIds.ListenerDisposing, "Disposing the listener.");
 
                     Stop();
                     DisposeInternal();
                 }
                 catch (Exception exception)
                 {
-                    Logger.LogError(0, exception, "Dispose");
+                    Logger.LogError(LoggerEventIds.ErrorInDispose, exception, "Dispose");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysListener.cs]: Also consistent ordering?
```suggestion
                    Logger.LogError(LoggerEventIds.ListenerDisposeError, exception, "Dispose");
```

File: src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs [hunk 18]
```diff
@@ -71,7 +71,7 @@ internal void SetProperty(HttpApiTypes.HTTP_SERVER_PROPERTY property, IntPtr inf
 
         internal void RegisterPrefix(string uriPrefix, int contextId)
         {
-            _logger.LogDebug("Listening on prefix: " + uriPrefix);
+            _logger.LogDebug(LoggerEventIds.RegisteringPrefix, "Listening on prefix: {0} " , uriPrefix);
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs]: Good formatting improvement.
```suggestion
            _logger.LogDebug(LoggerEventIds.RegisteringPrefix, "Listening on prefix: {0}" , uriPrefix);
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs [hunk 19]
```diff
@@ -130,7 +130,7 @@ private unsafe void FlushInternal(bool endOfRequest, ArraySegment<byte> data = n
                 if (!RequestContext.DisconnectToken.IsCancellationRequested)
                 {
                     // This is logged rather than thrown because it is too late for an exception to be visible in user code.
-                    Logger.LogError("ResponseStream::Dispose; Fewer bytes were written than were specified in the Content-Length.");
+                    Logger.LogError(LoggerEventIds.LessBytesThenExpected, "ResponseStream::Dispose; Fewer bytes were written than were specified in the Content-Length.");
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs]:
```suggestion
                    Logger.LogError(LoggerEventIds.FewerBytesThanExpected, "ResponseStream::Dispose; Fewer bytes were written than were specified in the Content-Length.");
```

**Review by Tratcher — CHANGES_REQUESTED** [review]
Much better. One more pass to make the names more consistent and you should be done.

File: src/Servers/HttpSys/src/HttpSysListener.cs (hunk 17 above)
Comment (kuns200) [reply src/Servers/HttpSys/src/HttpSysListener.cs]: made all events to look as Error.

**Comment by Tratcher** [comment]
Please move your Kestrel change to it's own branch and pull request. It's easier to review and merge distinct features.

File: src/Servers/HttpSys/src/LoggerEventIds.cs [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+        public static EventId FileSendAsyncError = new EventId(38, "FileSendAsyncError");
+        public static EventId FileSendAsyncCancelled = new EventId(39, "FileSendAsyncCancelled");
+        public static EventId FileSendAsyncErrorIgnored = new EventId(40, "FileSendAsyncErrorIgnored");
+        public static EventId WriteCancelled = new EventId(41, "WriteCancelled");
+        public static EventId ListenerStopping = new EventId(42, "ListenerStopping");
+        public static EventId ListenerStartError = new EventId(43, "ListenerStartError");
+        public static EventId DisconnectTriggered = new EventId(44, "DisconnectTriggered");
+        public static EventId ListenerStopError = new EventId(45, "ListenerStopError");
+        public static EventId ListenerDisposing = new EventId(46, "ListenerDisposing");
+
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/LoggerEventIds.cs]: Remove extra lines
```suggestion
```

**Review by Tratcher — APPROVED** [review]

File: src/Servers/HttpSys/src/LoggerEventIds.cs [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+        public static EventId FileSendAsyncCancelled = new EventId(39, "FileSendAsyncCancelled");
+        public static EventId FileSendAsyncErrorIgnored = new EventId(40, "FileSendAsyncErrorIgnored");
+        public static EventId WriteCancelled = new EventId(41, "WriteCancelled");
+        public static EventId ListenerStopping = new EventId(42, "ListenerStopping");
+        public static EventId ListenerStartError = new EventId(43, "ListenerStartError");
+        public static EventId DisconnectTriggered = new EventId(44, "DisconnectTriggered");
+        public static EventId ListenerStopError = new EventId(45, "ListenerStopError");
+        public static EventId ListenerDisposing = new EventId(46, "ListenerDisposing");
+    
+
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/LoggerEventIds.cs]:
```suggestion
```

File: src/Servers/HttpSys/src/LoggerEventIds.cs [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+        public static EventId FileSendAsyncErrorIgnored = new EventId(40, "FileSendAsyncErrorIgnored");
+        public static EventId WriteCancelled = new EventId(41, "WriteCancelled");
+        public static EventId ListenerStopping = new EventId(42, "ListenerStopping");
+        public static EventId ListenerStartError = new EventId(43, "ListenerStartError");
+        public static EventId DisconnectTriggered = new EventId(44, "DisconnectTriggered");
+        public static EventId ListenerStopError = new EventId(45, "ListenerStopError");
+        public static EventId ListenerDisposing = new EventId(46, "ListenerDisposing");
+    
+   
+
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/LoggerEventIds.cs]:
```suggestion
```

File: src/Servers/HttpSys/src/LoggerEventIds.cs [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/LoggerEventIds.cs]:
```suggestion
```

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**